File: .go/cli/kube/kube.go
/*
 * Copyright (c) 2025.
 * Created by Andy Pangaribuan (iam.pangaribuan@gmail.com)
 * https://github.com/apangaribuan
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 * All Rights Reserved.
 */

package kube

import (
	"squirrel/arg"
	"squirrel/util"

	"github.com/wissance/stringFormatter"
)

func CLI() {
	helpMessage := stringFormatter.FormatComplex(`
info : execute kubectl cli
usage: sq kube

{commands}
  pods     show pods information
  action   comprehensive kubectl execution
`, map[string]any{
		"commands": util.ColorBoldGreen("commands:"),
	})

	arg.Watch("sq kube", helpMessage, helpMessage).
		Add("pods", "", cliKubePods).
		Add("action", "", cliKubeAction).
		Exec()
}

File: src/cli/kube.rs
/*
 * Copyright (c) 2026.
 * Created by Andy Pangaribuan
 *
 * This product is protected by copyright and distributed under
 * licenses restricting copying, distribution and decompilation.
 * All Rights Reserved.
 */

use std::process::Command;

pub fn run(args: &[String]) {
    if args.is_empty() || args[0] == "--help" || args[0] == "-h" {
        println!("info : execute kubectl cli");
        println!("usage: sq kube");
        return;
    }

    let status = Command::new("kubectl").args(args).status();

    if let Err(err) = status {
        eprintln!("Error executing kubectl: {}", err);
        std::process::exit(1);
    }
}
